solarkraft/test/e2e/verify.test.ts

```diff
@@ -68,11 +68,39 @@ describe('verify', () => {
     })
 
     it('reports failure on deadlocking monitor, if another succeeding monitor is present', function (done) {
-        this.timeout(50000)
+        this.timeout(50_000)
         spawn(
             'solarkraft verify --home test/e2e/tla/ --txHash timelock --monitor test/e2e/tla/timelock_mon1_instrumented_fail.tla test/e2e/tla/timelock_mon2.tla'
         )
             .wait('[Fail]')
             .run(done)
     })
+
+    const txs = [
+        'df32f09514abedbf9d4b843fb8ad53de81d74b62914c4c9faf290760324abd0f',
+        '8f97b05ce36b9f2b2988150131cd8990bd018dfcace5250ad5df5e872af436bf',
+        '03a01ff64502f5d992bb28e784212235a5da55ebb68c75345f58364559cb5921',
+        '337365516cbe5f9287bc6b81428fb94be08d8e5d1ba580454d0907378c06c0da',
+        'ad3c4d82300b50972fdbc887c20bb7771c722a1c63bd44c8e8d4ddf4597f4143',
+        'e3dba60ada9fd7cb1d3490cf72b5e53e17b154e237aed2d2dcacfec6b0b1d883',
+        '762575ca214bd9cb85824356b799689bf4920aafcac24f790defac54b1a0cc9d',
+        '98c2468aad6626b59230b4fcc83d921897cbf76dc49f7bea13e46e443035a682',
+        'df673ab5399dcbd5ac434c0ac51c53e9b9367b1d2507ab7e4b8cebfe3e796499',
+        '101111c4a7087070366d035f59cc20b80215231955f9253c4ad85ff94f1dc61b',
+        '9c46c29a985ceeaffe567dc405a757805ff0496fe5740228abcb22b7aefa3e64',
+        '4da2e37effde690b6cf2c11ade5fc2f4e72971b37866f6d75fa2ea5728b6be6e',
+        'b39cf7d57120fa72bde0aec90f709551ff777132e1b5db89e2575ee942d8bb70',
+        '1bd53234a4eb4f9316f63bd77f4a2c191bded377ad3c53caeb7ed285f9d77d64',
+        '406d278860b5531dd1443532f3457c5daa288e8eb0007d2a8e2aa0127e87949e',
+    ]
+    txs.forEach(tx => {
+        it(`verifies the setter contract (tx ${tx})`, function (done) {
+            this.timeout(150_000)
+            spawn(
+                `solarkraft verify --home test/e2e/tla/ --txHash ${tx} --monitor test/e2e/tla/setter_mon.tla`
+            )
+                .wait('[OK]')
```

Comment:
is it `[OK]` or `OK`? Because I get errors when running the current tests implying this.

Reply:
<img width="940" alt="Screenshot 2024-05-23 at 15 05 32" src="https://github.com/freespek/solarkraft/assets/23355223/71fb9931-8530-4655-9d83-8683f606795b">

Reply:
I fixed this, but the tests still don't succeed:
<details>
<summary>Output</summary>
```
  list
    ✔ lists a single entry (112ms)

  verify
    ✔ errors on missing monitor (107ms)
    ✔ errors on no monitors given (107ms)
    ✔ errors on missing tx hash (110ms)
    1) reports success on okay monitor1
    2) reports success on okay monitor2
    3) reports success on two okay monitors
    4) reports failure on deadlocking monitor
    5) reports failure on deadlocking monitor, if another succeeding monitor is present
    6) verifies the setter contract (tx df32f09514abedbf9d4b843fb8ad53de81d74b62914c4c9faf290760324abd0f)
    7) verifies the setter contract (tx 8f97b05ce36b9f2b2988150131cd8990bd018dfcace5250ad5df5e872af436bf)
    8) verifies the setter contract (tx 03a01ff64502f5d992bb28e784212235a5da55ebb68c75345f58364559cb5921)
    9) verifies the setter contract (tx 337365516cbe5f9287bc6b81428fb94be08d8e5d1ba580454d0907378c06c0da)
    10) verifies the setter contract (tx ad3c4d82300b50972fdbc887c20bb7771c722a1c63bd44c8e8d4ddf4597f4143)
    11) verifies the setter contract (tx e3dba60ada9fd7cb1d3490cf72b5e53e17b154e237aed2d2dcacfec6b0b1d883)
    12) verifies the setter contract (tx 762575ca214bd9cb85824356b799689bf4920aafcac24f790defac54b1a0cc9d)
    13) verifies the setter contract (tx 98c2468aad6626b59230b4fcc83d921897cbf76dc49f7bea13e46e443035a682)
    14) verifies the setter contract (tx df673ab5399dcbd5ac434c0ac51c53e9b9367b1d2507ab7e4b8cebfe3e796499)
    15) verifies the setter contract (tx 101111c4a7087070366d035f59cc20b80215231955f9253c4ad85ff94f1dc61b)
    16) verifies the setter contract (tx 9c46c29a985ceeaffe567dc405a757805ff0496fe5740228abcb22b7aefa3e64)
    17) verifies the setter contract (tx 4da2e37effde690b6cf2c11ade5fc2f4e72971b37866f6d75fa2ea5728b6be6e)
    18) verifies the setter contract (tx b39cf7d57120fa72bde0aec90f709551ff777132e1b5db89e2575ee942d8bb70)
    19) verifies the setter contract (tx 1bd53234a4eb4f9316f63bd77f4a2c191bded377ad3c53caeb7ed285f9d77d64)
    20) verifies the setter contract (tx 406d278860b5531dd1443532f3457c5daa288e8eb0007d2a8e2aa0127e87949e)


  4 passing (3s)
  20 failing

  1) verify
       reports success on okay monitor1:

      AssertionError [ERR_ASSERTION]: expected "timelock_mon1.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -timelock_mon1.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  2) verify
       reports success on okay monitor2:

      AssertionError [ERR_ASSERTION]: expected "timelock_mon2.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -timelock_mon2.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  3) verify
       reports success on two okay monitors:

      AssertionError [ERR_ASSERTION]: expected "timelock_mon2.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -timelock_mon2.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  4) verify
       reports failure on deadlocking monitor:

      AssertionError [ERR_ASSERTION]: Non-empty queue on spawn exit.
[wait] unable to reproduce the transaction
[wait] [Fail]
      + expected - actual

      -[
      -  "[wait] unable to reproduce the transaction"
      -  "[wait] [Fail]"
      -]
      +[]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createUnexpectedEndError (node_modules/nexpect/lib/nexpect.js:322:10)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:225:19)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  5) verify
       reports failure on deadlocking monitor, if another succeeding monitor is present:

      AssertionError [ERR_ASSERTION]: expected "timelock_mon2.tla: Internal error, see below" to contain "[Fail]"
      + expected - actual

      -timelock_mon2.tla: Internal error, see below
      +[Fail]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  6) verify
       verifies the setter contract (tx df32f09514abedbf9d4b843fb8ad53de81d74b62914c4c9faf290760324abd0f):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  7) verify
       verifies the setter contract (tx 8f97b05ce36b9f2b2988150131cd8990bd018dfcace5250ad5df5e872af436bf):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  8) verify
       verifies the setter contract (tx 03a01ff64502f5d992bb28e784212235a5da55ebb68c75345f58364559cb5921):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  9) verify
       verifies the setter contract (tx 337365516cbe5f9287bc6b81428fb94be08d8e5d1ba580454d0907378c06c0da):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  10) verify
       verifies the setter contract (tx ad3c4d82300b50972fdbc887c20bb7771c722a1c63bd44c8e8d4ddf4597f4143):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  11) verify
       verifies the setter contract (tx e3dba60ada9fd7cb1d3490cf72b5e53e17b154e237aed2d2dcacfec6b0b1d883):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  12) verify
       verifies the setter contract (tx 762575ca214bd9cb85824356b799689bf4920aafcac24f790defac54b1a0cc9d):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  13) verify
       verifies the setter contract (tx 98c2468aad6626b59230b4fcc83d921897cbf76dc49f7bea13e46e443035a682):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  14) verify
       verifies the setter contract (tx df673ab5399dcbd5ac434c0ac51c53e9b9367b1d2507ab7e4b8cebfe3e796499):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  15) verify
       verifies the setter contract (tx 101111c4a7087070366d035f59cc20b80215231955f9253c4ad85ff94f1dc61b):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  16) verify
       verifies the setter contract (tx 9c46c29a985ceeaffe567dc405a757805ff0496fe5740228abcb22b7aefa3e64):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  17) verify
       verifies the setter contract (tx 4da2e37effde690b6cf2c11ade5fc2f4e72971b37866f6d75fa2ea5728b6be6e):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  18) verify
       verifies the setter contract (tx b39cf7d57120fa72bde0aec90f709551ff777132e1b5db89e2575ee942d8bb70):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  19) verify
       verifies the setter contract (tx 1bd53234a4eb4f9316f63bd77f4a2c191bded377ad3c53caeb7ed285f9d77d64):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)

  20) verify
       verifies the setter contract (tx 406d278860b5531dd1443532f3457c5daa288e8eb0007d2a8e2aa0127e87949e):

      AssertionError [ERR_ASSERTION]: expected "setter_mon.tla: Internal error, see below" to contain "[OK]"
      + expected - actual

      -setter_mon.tla: Internal error, see below
      +[OK]
      
      at new AssertionError (node:internal/assert/assertion_error:451:5)
      at createExpectationError (node_modules/nexpect/lib/nexpect.js:336:13)
      at flushQueue (node_modules/nexpect/lib/nexpect.js:239:21)
      at ChildProcess.<anonymous> (node_modules/nexpect/lib/nexpect.js:298:43)
      at ChildProcess.emit (node:events:519:28)
      at maybeClose (node:internal/child_process:1105:16)
      at Socket.<anonymous> (node:internal/child_process:457:11)
      at Socket.emit (node:events:519:28)
      at Pipe.<anonymous> (node:net:337:12)
```
</details>

Reply:
It's also failing for me. I guess, it's only working in the dev container?

Reply:
@Kukovec I think you'll have to run `verify` manually to see why it fails.
E.g., do you have Apalache at the expected path?

Reply:
I assume this was resolved by #79?